data/base/controller.php

```diff
@@ -147,50 +163,40 @@ public function register_processors() {
 	}
 
 	/**
-	 * Register internal endpoints.
+	 * Register index endpoint.
 	 */
-	protected function register_internal_endpoints() {
-		register_rest_route( $this->get_namespace(), '/' . $this->get_rest_base(), [
-			[
-				'methods' => WP_REST_Server::READABLE,
-				'callback' => array( $this, 'get_items' ),
-				'args' => [],
-				'permission_callback' => function ( $request ) {
-					return $this->get_permission_callback( $request );
-				},
-			],
+	protected function register_index_endpoint() {
+		$this->register_endpoint( new Endpoint\Index( $this ), [
+			'index' => true,
 		] );
 	}
 
 	/**
 	 * Register endpoint.
 	 *
-	 * @param string $endpoint_class
+	 * @param \Elementor\Data\Base\Endpoint $endpoint
+	 * @param array $args
 	 *
 	 * @return \Elementor\Data\Base\Endpoint
 	 */
-	protected function register_endpoint( $endpoint_class ) {
-		$endpoint_instance = new $endpoint_class( $this );
-
-		// TODO: Validate instance like in register_sub_endpoint().
-
-		$endpoint_route = $this->get_name() . '/' . $endpoint_instance->get_name();
-
-		$this->endpoints[ $endpoint_route ] = $endpoint_instance;
+	protected function register_endpoint( Endpoint $endpoint, $args = [] ) {
+		$command = $endpoint->get_full_command();
 
-		$command = $endpoint_route;
-		$format = $endpoint_instance::get_format();
+		// TODO: Remove - Backwards compatibility.
+		$endpoint_proxy = new Endpoint\Proxy( $endpoint );
 
-		if ( $command ) {
-			$format = $command . '/' . $format;
+		if ( ! empty( $args['index'] ) ) {
+			$this->index_endpoint = $endpoint_proxy;
 		} else {
-			$format = $format . $command;
+			$this->endpoints[ $command ] = $endpoint_proxy;
 		}
```

Comment:
https://stackoverflow.com/questions/4915441/manipulate-php-instanceof-operator-for-wrapper-class/38279539